CIP-DappConnector/CIP-dapp-connector.md

```diff
@@ -0,0 +1,269 @@
+}
+```
+
+* ProofGeneration - Wallet could not sign the data (e.g. does not have the secret key associated with the address)
+* AddressNotPK - Address was not a P2PK address and thus had no SK associated with it.
+* UserDeclined - User declined to sign the data
+* InvalidFormat - If a wallet enforces data format requirements, this error signifies that the data did not conform to valid formats.
+
+### PaginateError
+
+```
+type PaginateError = {|
+    maxSize: number,
+|};
+```
+{maxSize} is the maximum size for pagination and if the dApp tries to request pages outside of this boundary this error is thrown.
+
+### TxSendError
+
+```
+TxSendErrorCode = {
+	Refused: 1,
+	Failure: 2,
+}
+type TxSendError = {
+	code: TxSendErrorCode,
+	info: String
+}
+```
+
+* Refused - Wallet refuses to send the tx (could be rate limiting)
+* Failure - Wallet could not send the tx
+
+### TxSignError
+
+```
+TxSignErrorCode = {
+	ProofGeneration: 1,
+	UserDeclined: 2,
+}
+type TxSignError = {
+	code: TxSignErrorCode,
+	info: String
+}
+```
+
+* ProofGeneration - User has accepted the transaction sign, but the wallet was unable to sign the transaction (e.g. not having some of the private keys)
+* UserDeclined - User declined to sign the transaction
+
+
+
+## Initial API
+
+In order to initiate communication from webpages to a user's Cardano wallet, the wallet must provide the following javascript API to the webpage. A shared, namespaced `cardano` object must be injected into the page if it did not exist already. Each wallet implementing this standard must then create a field in this object with a name unique to each wallet containing a `wallet` object with the following methods. The API is split into two stages to maintain the user's privacy, as the user will have to consent to `cardano.walletName.enable()` in order for the dApp to read any information pertaining to the user's wallet.
+
+### wallet.enable(): Promise\<API>
+
+Errors: APIError
+
+This is the entrypoint to start communication with the user's wallet. The wallet should request the user's permission to connect the web page to the user's wallet, and if permission has been granted, the full API will be returned to the dApp to use. The wallet can choose to maintain a whitelist to not necessarily ask the user's permission every time access is requested, but this behavior is up to the wallet and should be transparent to web pages using this API. If a wallet is already connected this function should not request access a second time, and instead just return the `API` object.
+
+### wallet.isEnabled(): Promise\<bool>
+
+Errors: APIError
+
+Returns true if the dApp is already connected to the user's wallet, or if requesting access would return true without user confirmation (e.g. the dApp is whitelisted), and false otherwise. If this function returns true, then any subsequent calls to `wallet.enable()` during the current session should succeed and return the `API` object.
+
+### wallet.version: String
```

Comment:
I agree. @alessandrokonrad just FYI since you're already started implementing this - I'm not sure if you'll want to populate both fields or just have a breaking change. I hope it's okay to still make changes before we merge this CIP.